core-ignite/src/main/java/io/micronaut/ignite/intercept/GridfyAdvice.java

```diff
@@ -0,0 +1,91 @@
+package io.micronaut.ignite.intercept;
+
+import io.micronaut.aop.MethodInterceptor;
+import io.micronaut.aop.MethodInvocationContext;
+import io.micronaut.context.BeanContext;
+import io.micronaut.core.annotation.AnnotationValue;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.compute.ComputeTask;
+import org.apache.ignite.compute.gridify.Gridify;
+import org.apache.ignite.compute.gridify.GridifyArgument;
+import org.apache.ignite.compute.gridify.GridifyInterceptor;
+import org.apache.ignite.compute.gridify.aop.GridifyArgumentAdapter;
+import org.apache.ignite.compute.gridify.aop.GridifyDefaultTask;
+import org.apache.ignite.internal.util.typedef.F;
+import org.apache.ignite.internal.util.typedef.G;
+
+import javax.inject.Singleton;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.Optional;
+
+import static org.apache.ignite.IgniteState.STARTED;
+
+@Singleton
+public class GridfyAdvice implements MethodInterceptor<Object,Object> {
+
+    private final BeanContext beanContext;
+
+
+    public GridfyAdvice(BeanContext beanContext) {
+        this.beanContext = beanContext;
+    }
+
+    @Override
+    public int getOrder() {
+        return 0;
+    }
+
+    @Override
+    public Object intercept(MethodInvocationContext<Object, Object> context) {
+        Optional<AnnotationValue<Gridify>> opt = context.findAnnotation(Gridify.class);
+        if (!opt.isPresent()) {
+            return context.proceed();
+        }
+        String igniteInstanceName = opt.get().stringValue("igniteInstanceName").orElse(opt.get().stringValue("gridName").orElse(null));
+
+        if (F.isEmpty(igniteInstanceName)) {
+            throw new RuntimeException(new IgniteCheckedException("Grid is not locally started: " + igniteInstanceName));
+        }
+
+        if (G.state(igniteInstanceName) != STARTED) {
+            throw new RuntimeException(new IgniteCheckedException("Grid is not locally started: " + igniteInstanceName));
+        }
+
+        // Initialize defaults.
+        Method mtd = context.getTargetMethod();
```

Comment:
This will trigger the use of reflection, and should be avoided. All of the things you are after can be obtained through `getExecutableMethod()`